### DISCUSSION TITLE
I can't login with Wamp in local development environment

### DISCUSSION THREAD

**Discussion by tyl3r** [datenstrom/community#74] [Report a bug] [answered]
Just downloaded and installed in my local enviroment (WAMP) in http://test/yellow, i added the user in `user.ini` file:

```
admin@admin.com,$2y$10$Z6BmVdxwIopMGm0xeARtW.Brb9gLbVq8Aqz2.fptOyu85hOevOZbS,Yellow,en,/
```

I can't login, when i submit the form with the correct email and password (admin@admin.com | admin) nothing happen, the form just appear again...

If i submit the form with incorrect data works as expected:

```
Unauthorised
You are not authorised on this server. Please log in. 
```

Any idea to solve this, im missing something?

**Update:**
*I tested in Firefox and the login works but when i create a new page nothing is saved.., i only can edit the existing ones... And the link `edit this page` in Home doesn't work, for editing i must click `Edit page` on Header.
*In IE 11 i have the same problem as in Chrome, can't login!

Anything on my setup must be very wrong O_o, any idea?

**Comment by markseuffert** [comment]
Hi, your user configuration looks correct. Can you try out the [Yellow demo](https://datenstrom.se/yellow/demo/), can you log in there?

**Comment by tyl3r** [comment]
Yes, i tried Yellow online in Chrome before downloading it.

Testing again: I clicked now in the url and im already logged in, can edit & save all page/posts. I logout and login again, works as expected.

**Comment by markseuffert** [comment]
It could be a browser setting. When you are logging in with the correct email+password a cookie is created from the server. Perhaps the web browser was not allowed to set cookies? 

Thanks for your detailed error report, appreciate it. Please let me know if it happens again or you find more problems.

**Comment by tyl3r** [comment]
Sorry my english is not so good xD, you understanded: now it works, but no, **Testing again:** i mean online in your demo.

Don't know what cause this behavior in my system but i can't use Yellow in local, i just don't understand why don't work :/, my server is Wamp x64 and i tested with exit other cms like: anchor, bolt, ghost, mecha, october and wolf.

Now i don't have time to do more test, but in the next days i'll try Yellow again at local and in my online server, if i discover the problem i'll communicate you here.

**Comment by markseuffert** [comment]
Could you find something out? I am happy to help, just contact me if you want me to look at it.

**Comment by FabienPean** [comment]
Hi there,
Same problem for me. Downloaded from your website and installed on my wamp x64. Also tried with Opera (latest version or close) and Internet explorer. The login box is looping without giving me access to edit mode.
However login/edit mode works on your demo page.

**Comment by markseuffert** [comment]
Thanks, so far I don't know what's happening. It could be related to Windows or PHP version.

Can you create me a login and contact me? If that's okay with you, I would like to have a look at it.

**Comment by FabienPean** [comment]
PHP 5.5.12 (cli) (built: Apr 30 2014 11:20:58)
Copyright (c) 1997-2014 The PHP Group
Zend Engine v2.5.0, Copyright (c) 1998-2014 Zend Technologies
    with Xdebug v2.2.5, Copyright (c) 2002-2014, by Derick Rethans

As for the account, sorry my setup is only local.

I did upload yellow it on my remote shared hosting server and there was no problem with login/edit.

**Comment by markseuffert** [comment]
Can you [create a user account at the command line](https://github.com/datenstrom/yellow-extensions/tree/master/source/edit#how-to-create-a-user-account) or do you get an error?

**Comment by FabienPean** [comment]
Yes I can create an user in powershell: "Yellow user: User account created" And it appears in the user.ini file

**Comment by markseuffert** [comment]
Thanks Fabien. Let's try to find out how to reproduce the problem. 

Please post your WAMP and Windows version, if you have this problem.

**Comment by FabienPean** [comment]
WampServer Version 2.5 x64
Windows 7 Professional x64

**Comment by nibreh** [comment]
Hello - I don't use WAMP, but I try it on Windows 7 professional x64

What I've done:
- Install [WAMP 2.5 64b from sourceforge](http://sourceforge.net/projects/wampserver/files/WampServer%202/Wampserver%202.5/) (APACHE 2.4.9 / PHP 5.5.12)
- active rewrite_module
- Install yellow in c://wamp/www
- go to http://localhost/yellow with Firefox

Login and create page is working fine.

_This issue is a mystery ..._

I try UwAmp too, it's a portable wamp server. And Yellow it's working as well. 

> http://www.uwamp.com/en/

**Comment by markseuffert** [comment]
Thanks @nibreh, I followed your instructions.

I installed WAMP on a computer and found something out today. It looks like some web browsers don't store cookies for `localhost`. You need to use a longer hostname like `local.server` to make it work with all browsers. Let me see if I can find a bugfix for localhost... which is the default for WAMP.

Bug reproduced with:

WampServer Version 2.5 x64
Windows 7 Professional x64
Google Chrome 43.0

**Comment by markseuffert** [comment, marked as answer]
It should be working now. 

You should be able to go to `http://localhost/yellow/` on your Windows computer and log in. I tested it with  Chrome and Internet Explorer. Please let me know if this works for you. Thanks.

**Comment by humantex** [comment]
Sorry to rehash this issue, but I've just done a fresh yellow install on win7 64, w/wamp v1.8.0 (php 5.4.4)...

I get no login prompt at all, and no command line response or even an error when I try adding a new user. I find no cookies created by yellow under any possible server address, i.e., 'localhost/yellow', '127.0.0.1/yellow', '192.168.0.11/yellow' (my internal IP), or even using '\myMachineName/yellow'. I can only see the default pages using any of these, and the not-authorized error when I click on 'edit'.

Is there any url string that will force a login, or any other way to get into admin mode?

**Comment by humantex** [comment]
I should also note that I've used this setup for years with lots of other scripts in subfolders and have never had any similar web server issues...

**Comment by markseuffert** [comment]
Can you try the latest version of WAMP and check if it still happens? ~~I suspect PHP 5.4.4 is the problem.~~

**Comment by humantex** [comment]
Unfortunately, I doubt I can do an overall upgrade easily. I've upgraded individual web components from the standard wamp package to mirror the versions on the server I publish to. 

I'll need to go through the hosting server setup to see what changes are possible there, or what it may have been recently updated to. I can't set up my own local server to anything that doesn't sync to whats on my live server. I have other projects to worry about and can't make wholesale changes just to test yellow.

As it is now... Apache 2.4.2, php 5.4.4, , mysql 5.0.10, so I need to investigate further.

I was hoping that yellow would negate me even having to worry about these kinds of issues, since the intent was to use it as a static html generator. Develop what I needed locally, then post/sync the generated pages online.

**Comment by humantex** [comment]
The production server has php 5.3.7. It's stable/hardened and, for now, version locked... so 5.4.4 is as far as I can go locally and stay compatible.

Looks like yellow isn't a possibility for me if it's a php versioning issue.

**Comment by markseuffert** [comment]
I understand what you mean, never change a running system. Do you have the possibility to install latest version of WAMP and Yellow on another Windows computer? I like to eliminate that this is a WAMP issue.

**Comment by markseuffert** [comment]
I tested Yellow 0.6.1, PHP 5.4.4 and it works fine. Please open a new issue if the problem remains.
